tools/tpm2_rsadecrypt.c

```diff
@@ -123,6 +148,9 @@ static bool on_option(char key, char *value) {
         ctx.flags.o = 1;
         break;
     }
+    case 's':
+        ctx.scheme.scheme = str_to_pad(value);
```

Comment:
tpm2_alg_util.c: tpm2_alg_util_from_optarg(value, tpm2_alg_util_flags_enc_scheme) should do this and all error checking.

Reply:
I looked at using that but I wasnt overly happy
-We would need tpm2_alg_util_flags_misc | tpm2_alg_util_flags_enc_scheme
-We have to filter out ecdh (it's in the tpm2_alg_util_flags_enc_scheme group but not a padding scheme option)
If that's still ok, I will make the changes

Reply:
apply this and work it up:
```diff
diff --git a/lib/tpm2_alg_util.c b/lib/tpm2_alg_util.c
index a3f15193cca8..4ec7e5e370a1 100644
--- a/lib/tpm2_alg_util.c
+++ b/lib/tpm2_alg_util.c
@@ -97,8 +97,8 @@ static void tpm2_alg_util_for_each_alg(alg_iter iterator, void *userdata) {
         { .name = "ecschnorr", .id = TPM2_ALG_ECSCHNORR, .flags = tpm2_alg_util_flags_sig },
 
         // Assyemtric Encryption Scheme
-        { .name = "oaep", .id = TPM2_ALG_OAEP, .flags = tpm2_alg_util_flags_enc_scheme },
-        { .name = "rsaes", .id = TPM2_ALG_RSAES, .flags = tpm2_alg_util_flags_enc_scheme },
+        { .name = "oaep", .id = TPM2_ALG_OAEP, .flags = tpm2_alg_util_flags_enc_scheme | tpm2_alg_util_flags_rsa_scheme},
+        { .name = "rsaes", .id = TPM2_ALG_RSAES, .flags = tpm2_alg_util_flags_enc_scheme | tpm2_alg_util_flags_rsa_scheme },
         { .name = "ecdh", .id = TPM2_ALG_ECDH, .flags = tpm2_alg_util_flags_enc_scheme },
 
 
@@ -123,7 +123,7 @@ static void tpm2_alg_util_for_each_alg(alg_iter iterator, void *userdata) {
         { .name = "keyedhash", .id = TPM2_ALG_KEYEDHASH, .flags = tpm2_alg_util_flags_base },
 
         // Misc
-        { .name = "null", .id = TPM2_ALG_NULL, .flags = tpm2_alg_util_flags_misc },
+        { .name = "null", .id = TPM2_ALG_NULL, .flags = tpm2_alg_util_flags_misc | tpm2_alg_util_flags_rsa_scheme},
     };
 
     size_t i;
diff --git a/lib/tpm2_alg_util.h b/lib/tpm2_alg_util.h
index e3783dfccab6..2caa72e22e3e 100644
--- a/lib/tpm2_alg_util.h
+++ b/lib/tpm2_alg_util.h
@@ -49,6 +49,7 @@ enum tpm2_alg_util_flags {
     tpm2_alg_util_flags_base        = 1 << 8,
     tpm2_alg_util_flags_misc        = 1 << 9,
     tpm2_alg_util_flags_enc_scheme  = 1 << 10,
+    tpm2_alg_util_flags_rsa_scheme  = 1 << 11,
     tpm2_alg_util_flags_any         = ~0
 };
```

Reply:
TY